### PR TITLE
[SVCS-595] Clean up existing type annotation failures

### PR DESCRIPTION
**Pull request by TomBaxter**
## Ticket
[SVCS-595](https://openscience.atlassian.net/browse/SVCS-595)

## Purpose
Update box and google providers to pass mypy

## Changes
Fix type hints and/or signatures to pass mypy
Ignore two lines I was unable to resolve

## Side effects

None

## QA Notes

Only GoogleDrive and Box providers touched. 

- Box:
-- Intra copy/move
-- Folder metadata
- GoogleDrive:
-- Revisions

### CODE REVIEW

**Comment by Johnetordoff** [comment]
LGTM! Worth noting however that there are a few `# type: ignore` statements [like here](https://github.com/CenterForOpenScience/waterbutler/pull/314/files#diff-dc1d5e76ff191c30934d4e6726a96e14R183) on google drive that seems to serve no purpose, but that's not really in scope.

**Comment by cslzchen** [comment]
@Johnetordoff Thanks for the review. These are removed in https://github.com/CenterForOpenScience/waterbutler/pull/276/files.

File: waterbutler/providers/box/provider.py [hunk 1]
```diff
@@ -501,7 +500,8 @@ def _build_upload_url(self, *segments, **query):
             box_path = await self.validate_path(child.path)
             await self.delete(box_path)
 
-    async def _intra_move_copy_metadata(self, path, data: dict) -> BaseBoxMetadata:
+    async def _intra_move_copy_metadata(self, path, data: dict) -> typing.Tuple[BaseBoxMetadata,
```
Comment (cslzchen) [inline waterbutler/providers/box/provider.py]: Is there a reason that `path` does not have a type specified?  Is it a dropbox path (if there is one), waterbutler path or a string?

Comment (cslzchen) [reply waterbutler/providers/box/provider.py]: Update: `path` is of type `WaterButlerPath`.

**Pull request closed without merging**